### PR TITLE
Scaling bugs fixes

### PR DESCRIPTION
**Pull request by zakv**
There are a few bugs with scalings, which I've patched in this PR.

The first one is that the uncertainties were scaled the wrong way in `GuassianProcessLearner`. When normalizing costs, the `cost_scaler` divides them by `cost_scaler.scale_` (see [here](https://github.com/scikit-learn/scikit-learn/blob/0fb307bf39bbdacd6ed713c00724f8f871d60370/sklearn/preprocessing/_data.py#L807)) and when rescaling costs back to their original range it multiplies the scaled cost by `cost_scaler.scale_` (see [here](https://github.com/scikit-learn/scikit-learn/blob/0fb307bf39bbdacd6ed713c00724f8f871d60370/sklearn/preprocessing/_data.py#L845)). The same should be done for the uncertainties, but previously the multiplication and division were switched. This PR corrects that issue. Furthermore the uncertainties in `GaussianProcessVisualizer.return_cross_sections()` weren't rescaled back, and this has been corrected as well.

The second issue is more minor. The value of `cross_parameter_arrays` in `GaussianProcessLearner.return_cross_sections()` and `NeuralNetLearner.return_cross_sections()` was scaled previously, but does not need to be scaled. When it is created, its values go between `self.min_boundary` and `self.max_boundary` which aren't scaled, so there's no scaling to undo. This was likely a small mix-up where the scaling was meant to be applied to `uncertainty_arrays` but was accidentally applied to `cross_parameter_arrays` instead, and then that mix-up was copy/pasted to `NeuralNetVisualizer.return_cross_sections()` . Fortunately the value returned for `cross_parameter_arrays` isn't used anywhere in M-LOOP, so it didn't cause any other issues.

Changes proposed in this pull request:

- Correct uncertainty scaling in `GaussianProcessLearner` and `GaussianProcessVisualizer`
- Correct `cross_parameter_arrays` scaling in `GaussianProcessVisualizer` and `NeuralNetVisualizer`

@qctrl/support @charmasaur

### CODE REVIEW

**Comment by zakv** [comment]
I should also mention that there are a lot of scalings done in `neuralnet.py`. I haven't tested any of those, but they all looked correct when I read through them!

**Review by charmasaur — APPROVED** [review]
Nice catch!
